### PR TITLE
Feature/tasks api

### LINKED ISSUE #6: [Design] Homework storage

### LINKED ISSUE #7: [Dev] Set up a development environment

### LINKED ISSUE #18: [Dev] Homeworks
## Disclaimer 
Please note that the architectural approach used in this project is intentionally complex and not reflective of what would typically be necessary for a task of this scope. This complexity was adopted solely for educational purposes to provide hands-on experience with certain design patterns and advanced features. We acknowledge that this might not be the optimal approach for the given task and appreciate your understanding and feedback.

## 1
Create BLL (business logic layer) and DAL (data-access layer) for Tasks Service. Any implementation of the layered architecture can be used. Layers should be logically separated (via separate folders/namespaces).
### Functional Requirements:
1. Single entity – Task
2. The task has a unique ID which is maintained (generated) on the client side.
3. The following actions should be supported:
   a. Get list of items.
   b. Add a task.
    c. Remove task

Each item contains the following info:
Id – required, ID of the item in an external system
Name – required, plain text.
Description – required, plain text.
URL – optional, plain text.

### Non-functional Requirements (NFR):
1. Testability
2. Extensibility

### Constraints:
1. NoSQL database for persistence layer (for example - https://www.litedb.org/).
2. Layers should be logically separated.

## 2
Create BLL (business logic layer) and DAL (data-access layer) for Homework Service. You must follow Clean Architecture with physical layers separation (via separate DLLs).

### Constraints:
1. SQL database for persistence layer (for example - Microsoft SQL Server Database File).
2. Layers should be physically separated.

### Non-functional Requirements (NFR):
1. Testability
2. Extensibility

### Functional Requirements:
1. Key entities: Homework
2. The following operations are allowed for Homework: get/list/add/update/delete.

### PR DESCRIPTION
**Pull request by Ungerfall**
---
# Tasks API
**Title:** Implement tasks API with Cosmos DB integration

**Description:**
This pull request introduces the tasks API along with Cosmos DB integration. Key changes include:

- **API presentation Layer**
- **Business Logic Layer**
- **Data Access Layer**:
  - Implemented cosmos db repository
---

# Homeworks API
**Title:** Implement homeworks API with Clean Architecture

**Description:**
This pull request introduces the homeworks API. Key changes include:

- **Web API presentation Layer**
- **Application layer**
- **Infrustructure layer**
- **Core (Domain) layer**
---

### CODE REVIEW

File: src/tasks-api/BusinessLayer/ILearningTaskService.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿using EKids.Chatbot.Tasks.DataAccessLayer.Entities;
+
+namespace EKids.Chatbot.Tasks.BusinessLayer;
+public interface ILearningTaskService
+{
+    Task AddTask(LearningTask task, Guid courseId, CancellationToken cancellation);
```
Comment (Hleb-s) [inline src/tasks-api/BusinessLayer/ILearningTaskService.cs:9]: Add Summary please (And in other Interfaces too)

**Review by Hleb-s — COMMENTED** [review]
Please do the unit test coverage

File: src/tasks-api/BusinessLayer/ILearningTaskService.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿using EKids.Chatbot.Tasks.DataAccessLayer.Entities;
+
+namespace EKids.Chatbot.Tasks.BusinessLayer;
```
Comment (Hleb-s) [inline src/tasks-api/BusinessLayer/ILearningTaskService.cs:3]: It is necessary to select layers for different projects

File: src/tasks-api/BusinessLayer/ILearningTaskService.cs (hunk 1 above)
Comment (Ungerfall) [reply src/tasks-api/BusinessLayer/ILearningTaskService.cs:9]: Added summaries

File: src/tasks-api/BusinessLayer/ILearningTaskService.cs (hunk 2 above)
Comment (Ungerfall) [reply src/tasks-api/BusinessLayer/ILearningTaskService.cs:3]: Moved layers to folders with separate projects

**Review by Hleb-s — APPROVED** [review]

**Pull request merged**